modules/png-extraction/ImageExtractor.py

```diff
@@ -273,142 +288,226 @@ def fix_mismatch(with_VRs=['PN', 'DS', 'IS']):
+        mappings ="{}/maps/mapping_{}.csv".format(output_directory,i)
+        fm = open(mappings, "w+")
+        filemapping = 'Original DICOM file location, PNG location \n'
+        fm.write(filemapping)
+        # add a check to see if the metadata has already been extracted
+        #%%step through whole file list, read in file, append fields to future dataframe of all files
+        headerlist = []
+        #start up a multi processing pool
+        #for every item in filelist send data to a subprocess and run extract_headers func
+        #output is then added to headerlist as they are completed (no ordering is done)
         with Pool(core_count) as p:
-            res = p.imap_unordered(extract_images,range(len(filedata)))
-            for out in res:
-                (fmap,fail_path,err) = out
+            res= p.imap_unordered(extract_headers, enumerate(chunk))
+            for i,e in enumerate(res):
+                headerlist.append(e)
+        data = pd.DataFrame(headerlist)
+        logging.info('Chunk ' + str(i) + ' Number of fields per file : ' + str(len(data.columns)))
+        #%%find common fields
+        #make dataframe containing all fields and all files minus those removed in previous block
+        #%%export csv file of final dataframe
+        export_csv = data.to_csv(csv_destination, index = None, header=True)
+        fields=data.keys()
+        count = 0 #potential painpoint
+        #writting of log handled by main process
+        if print_images:
+            logging.info("Start processing Images")
+            filedata = data
+            total = len(chunk)
+            stamp = time.time()
+            for i in range(len(filedata)):
+                res = extract_images(filedata, i, png_destination, flattened_to_level, failed, is16Bit)
+                # (fmap,fail_path,err) = out
+                fmap = res[1]
+                fail_path = res[0]
+                err = res[-1]
                 if err:
                     count +=1
                     copyfile(fail_path[0],fail_path[1])
                     err_msg = str(count) + ' out of ' + str(len(chunk)) + ' dicom images have failed extraction'
                     logging.error(err_msg)
                 else:
                     fm.write(fmap)
-    fm.close()
-    logging.info('Chunk run time: %s %s', time.time() - t_start, ' seconds!')
-
-
-logging.info('Generating final metadata file')
-
-#identify the 
-col_names = dict()
-all_headers = dict()
-
-metas = glob.glob( "{}*.csv".format(meta_directory))
-#for each meta  file identify the columns that are not na's for at least 10% (metadata_col_freq_threshold) of data 
-for meta in metas:
-    m = pd.read_csv(meta,dtype='str')
-    d_len = m.shape[0]
-    for e in m.columns:
-        if np.sum(m[e].isna()) < (1-metadata_col_freq_threshold)*d_len: # Column e is populated in at least 10% of rows (i.e. isNaN in <90% of rows)
-            if e in col_names:
-                col_names[e] += 1
+            '''
```

Comment:
Is this code segment commented out? Why?

Reply:
no, it has been moved down a bit

Reply:
oh ok.